[MI-3823]: Added the test cases for linking and unlinking channel on MM. (#23)
Brightscout/mattermost-test-management: data/test-cases/plugins/ms-teams-sync/ui-integration/Linking_channel.md

---
# (Required) Ensure all values are filled up
name: "Linking a channel on MM"
status: Active
priority: Normal
folder: ui-integration
authors: "@arush-vashishtha"
team_ownership: []
priority_p1_to_p4: P2 - Core Functions (Do core functions work?)

# (Optional)
location: null
component: null
tags: []
labels: []
tested_by_contributor: null

# (Optional) Test type and tools
cypress: null
detox: null
mmctl: null
playwright: null
rainforest: []
manual_test_environments: []

# Do not change
id: null
key: null
created_on: null
last_updated: null
case_hashed: null
steps_hashed: null
---

**Step 1**

1. Click on the `Link a channel` button in the RHS of MS Teams sync plugin on MM.
2. Select the desired `Mattermost channel`, the desired `Microsoft team` and a desired `Microsoft channel` for the selected MS team in the `Link a channel` modal.

> 3. Click on `Link channels` button in the `Link a channel` dialog.

Reviewer: Should we also write a precondition for view i.e., when no channels are linked?
For point 3, modal or dialog?
we can also link a modal through add button in the rhs, also the modal opens when we run slash command `/msteams-sync link`.
```suggestion
1. Click on the `Link a channel` button in the RHS of MS Teams sync plugin on MM.
2. Search and select the desired `Mattermost channel`, the desired `Microsoft team` and a desired `Microsoft channel` for the selected MS team in the `Link a channel` modal.
3. Click on `Link channels` button in the `Link a channel` dialog.
```

**Step 2**

1. Click on the `Link a channel` button in the RHS of MS Teams sync plugin on MM.

> 2. Click on the `cancel` or `x` button in the `Link a channel` dialog.

Reviewer: Do we need to add a test case for the disabled button of the link channel and MS Teams channel selection when all the fields are not filled and MS Teams teams are not selected respectively?

**Expected**

The desired channel should get linked and a toast message should be visible in the RHS of MS Teams sync plugin for successfull connection with updation in the list of linked channels on MM.

> After step 2, the `Link a channel` dialog should close and the RHS should persist it's state.

Reviewer: ```suggestion
The desired channel should get linked and a toast message should be visible in the RHS of the MS Teams sync plugin for successful linking with an update in the list of linked channels on MM.
After step 2, the `Link a channel` dialog should close and the RHS should persist its state.
```

**Step 3**

1. Click on the `Link a channel` button in the RHS of the MS Teams sync plugin on MM.
2. Click on the `Link channels` button in the `Link a channel` dialog.

**Step 4**

1. Click on the `Link a channel` button in the RHS of the MS Teams sync plugin on MM.
2. Click on the dropdown to select the desired `Microsoft channel` when the `Mattermost channel` and `Microsoft team` are not selected in the `Link a channel` dialog.

**Step 5**
1. Enter the slash command `/msteams-sync link` in the channel on MM.
2. Search and select the desired `Mattermost channel`, the desired `Microsoft team` and a desired `Microsoft channel` for the selected MS team in the `Link a channel` dialog.
3. Click on the `Link channels` button in the `Link a channel` dialog.

**Expected**

The desired channel should get linked and a toast message should be visible in the RHS of MS Teams sync plugin for successfull linking with an update in the list of linked channels on MM.
After step 2, the `Link a channel` dialog should close and the RHS should persist its state.
After step 3, the `Link channels` button should be disabled in the `Link a channel` dialog.

> After step 4, the dropdown list should be disabled for the `Microsoft channel` when the `Mattermost channel` and `Microsoft team` are not selected in the `Link a channel` dialog.

Reviewer: ```suggestion
After step 4, the dropdown list should be disabled for the `Microsoft Teams channel` when the `Microsoft Teams team` is not selected in the `Link a channel` dialog.
```

> After step 5, the desired channel should get linked and a toast message should be visible in the RHS of MS Teams sync plugin for successfull linking with an update in the list of linked channels on MM.

Reviewer: ```suggestion
After step 5, the desired channel should get linked and a toast message should be visible in the RHS of the MS Teams sync plugin for successful linking with an update in the list of linked channels on MM.
```

Reviewer: toast will not appear while linking through slash command
